homeassistant/components/watchyourlan/sensor.py

```diff
@@ -0,0 +1,136 @@
+) -> None:
+    """Set up the WatchYourLAN sensors."""
+    coordinator: WatchYourLANUpdateCoordinator = entry.runtime_data
+
+    entities: Sequence[SensorEntity | BinarySensorEntity] = [
+        WatchYourLANOnlineStatusBinarySensor(coordinator, device)
+        if description.key == "online_status"
+        else WatchYourLANGenericSensor(coordinator, device, description)
+        for device in coordinator.data
+        for description in ENTITY_DESCRIPTIONS
+    ]
+
+    async_add_entities(entities)
+
+
+class WatchYourLANOnlineStatusBinarySensor(
+    CoordinatorEntity[WatchYourLANUpdateCoordinator], BinarySensorEntity
+):
+    """Binary sensor to represent online/offline status."""
+
+    def __init__(
+        self,
+        coordinator: WatchYourLANUpdateCoordinator,
+        device: dict[str, Any],
+    ) -> None:
+        """Initialize the binary sensor for online/offline state."""
+        super().__init__(coordinator)
+        self.device = device
+        self._attr_unique_id = f"{self.device.get('Mac')}_online_status"
+        mac_address = self.device["Mac"]
+        self._attr_device_info = DeviceInfo(
+            connections={(CONNECTION_NETWORK_MAC, mac_address)},
+            name=self.device.get("Name")
+            or f"WatchYourLAN {self.device.get('ID', 'Unknown')}",
+            manufacturer=self.device.get("Hw", "Unknown Manufacturer"),
+            model="WatchYourLAN Device",
+        )
+
+    @property
+    def is_on(self) -> bool:
+        """Return true if the device is online."""
+        return self.device.get("Now", 0) == 1
+
+    @property
+    def name(self) -> str:
+        """Return the name of the sensor."""
+        return (
+            self.device.get("Name")
+            or f"WatchYourLAN {self.device.get('ID', 'Unknown')} Online Status"
+        )
+
+    @property
+    def device_class(self) -> BinarySensorDeviceClass | None:
+        """Return the class of this device, which is 'connectivity'."""
+        return BinarySensorDeviceClass.CONNECTIVITY
+
+
+class WatchYourLANGenericSensor(
+    CoordinatorEntity[WatchYourLANUpdateCoordinator], SensorEntity
+):
+    """Generic WatchYourLAN sensor for different data points."""
+
+    def __init__(
+        self,
+        coordinator: WatchYourLANUpdateCoordinator,
+        device: dict[str, Any],
+        description: SensorEntityDescription,
+    ) -> None:
+        """Initialize the sensor."""
+        super().__init__(coordinator)
+        self.device = device
+        self.entity_description = description
+        self._attr_unique_id = f"{self.device.get('ID')}_{description.key}"
+        self._attr_device_info = DeviceInfo(
+            connections={(CONNECTION_NETWORK_MAC, self.device["Mac"])},
+            name=self.device.get("Name")
+            or f"WatchYourLAN {self.device.get('ID', 'Unknown')}",
+            manufacturer=self.device.get("Hw", "Unknown Manufacturer"),
```

Comment:
```suggestion
            default_name=self.device.get("Name")
            or f"WatchYourLAN {self.device.get('ID', 'Unknown')}",
            default_manufacturer=self.device.get("Hw", "Unknown Manufacturer"),
```
This way if one of the devices is a device that is integrated via HA, that name takes priority over this one